### PR TITLE
[WIP] server: improve code coverage

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 

Improve code coverage of `server`.

### What is changed and how it works?

Add more tests for `server` componement

### Check List 

Tests 

 - Unit test

Code changes

None

Side effects

None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: server/http_status.go [hunk 1]
```diff
@@ -178,17 +177,18 @@ func (s *Server) startHTTPServer() {
 			serveError(w, http.StatusInternalServerError, fmt.Sprintf("Create zipped %s fail: %v", "profile", err))
 			return
 		}
-		if err := rpprof.StartCPUProfile(fw); err != nil {
```
Comment (zz-jason) [inline server/http_status.go]: why commenting these codes out?

**Comment by zz-jason** [comment]
@Deardrops Could you post the code coverage improvements?

**Comment by Deardrops** [comment]
@zz-jason This PR is still WIP, I will close this PR utill it finish.

**Pull request closed without merging**